src/main/java/duke/tasks/Deadline.java

```diff
@@ -0,0 +1,55 @@
+package duke.tasks;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+/** This class represents a Deadline. */
+public class Deadline extends Task {
+    protected LocalDate date;
+    protected LocalTime time;
+
+    /**
+     * Constructs a Deadline object associated with a description and a date.
+     * @param description The description of the task.
+     * @param date The date of the deadline.
+     */
+    public Deadline(String description, LocalDate date) {
+        super(description);
+        this.date = date;
+    }
+
+    /**
+     * Constructs a Deadline object associated with a description, a date and a time.
+     * @param description The description of the task.
+     * @param date The date of the deadline.
+     * @param time The time of the deadline.
+     */
+    public Deadline(String description, LocalDate date, LocalTime time) {
```

Comment:
I like how you have variations for users to input their Deadline and Event tasks! 👍